src/core/plugins/core/plugin.js

```diff
@@ -203,19 +203,21 @@ class CorePlugin extends Plugin {
             }, 10);
           })
             .then(downloadedFilePath => {
-              fs.ensureDir(
-                path.join(this.cachePath, this.props.config.codename, group)
-              ).then(() =>
-                fs.copyFile(
-                  downloadedFilePath,
-                  path.join(
-                    this.cachePath,
-                    this.props.config.codename,
-                    group,
-                    file.name
-                  )
+              return fs
+                .ensureDir(
+                  path.join(this.cachePath, this.props.config.codename, group)
                 )
-              );
+                .then(() =>
+                  fs.copyFile(
```

Comment:
I think this returns a `Promise` as well that you should to return to get the desired chaining behaviour? Otherwise it's probably safer to use `copyFileSync`?

Reply:
This returns, but only inside the scope of the `fs.ensureDir`->`fs.copyFile` chain.